storage/badger/blocks.go

```diff
@@ -65,7 +65,7 @@ func (b *Blocks) retrieveTx(blockID flow.Identifier) func(*badger.Txn) (*flow.Bl
 
 // Store ...
 func (b *Blocks) Store(block *flow.Block) error {
-	return operation.RetryOnConflictTx(b.db, transaction.Update, b.StoreTx(block))
+	return operation.TerminateOnFullDisk(operation.RetryOnConflictTx(b.db, transaction.Update, b.StoreTx(block)))
```

Comment:
Given we want all node roles to have the same behaviour in this case, I'd suggest we move the check for the out-of-disk error to a lower level. Currently, if we add a new storage method or resource, we need to remember to wrap insertion operations in `TerminateOnFullDisk`. If we put the logic in the common storage insertion methods, all current and future insertions will have the same behaviour. 

`batchInsert`, `insert`, and `update` in [this file](https://github.com/onflow/flow-go/blob/add2b9d79ace88740e9a4e3c1d19f4d2af36aa19/storage/badger/operation/common.go#L20) are the three base-level methods which can insert data to the Badger DB.

Reply:
> this is the implementation of the "light touch" idea

The intention there was to use the "light touch" approach only if we want different behaviour in the case of out-of-disk, depending on the caller. But to always have the same behaviour to handle out-of-disk, it's quite a bit simpler to put the check in those three lower-level storage methods. 

> also note that some disk writes for transactions also seem to happen in [this other file](https://github.com/onflow/flow-go/blob/710e4bf266d53882fa407374925ba23d9846655c/storage/badger/transaction/tx.go)

I think this file should be OK without changes. This file is creating and committing Badger transactions, but not directly inserting data itself. That would happen within the [functional argument](https://github.com/onflow/flow-go/blob/710e4bf266d53882fa407374925ba23d9846655c/storage/badger/transaction/tx.go#L28), and ultimately would be using those 3 insertion methods mentioned.